azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/implementation/GenericResourcesImpl.java

```diff
@@ -35,29 +37,37 @@
         ResourceManager>
     implements GenericResources {
 
-    GenericResourcesImpl(ResourceManager resourceManager) {
+    private PagedListConverter<GenericResourceExpandedInner, GenericResource> converter;
+
+    GenericResourcesImpl(final ResourceManager resourceManager) {
         super(resourceManager.inner().resources(), resourceManager);
+        converter = new PagedListConverter<GenericResourceExpandedInner, GenericResource>() {
+            @Override
+            public Observable<GenericResource> typeConvertAsync(GenericResourceExpandedInner inner) {
+                return Observable.just((GenericResource) new GenericResourceImpl(inner.name(), inner, manager()));
+            }
```

Comment:
No. This losses information.

If service do this intentionally, the right approach is to had `GenericResource` wraps `GenericResourceExpandedInner` (when by get, there additional properties be null).

Reply:
It might be OK if we determined not to maintain it anymore...

Reply:
Maybe it is OK.

Even if we later need to expose createdTime etc., we can hack by first check inner instanceof GenericResourceExpandedInner , if yet, cast and return value, if not, return null.

Reply:
Now I'm not sure if we should expose filter/expand in APIs. For the solution, I agree we can do in that way by checking instanceof.

Reply:
Generally speaking, we should expose as overloaded method. But for track1, I think we will not expose new ones, until requested by customer.

For track2, I think the question is whether we expose a simple string (and let user compose it based on description), or try standardize it a bit.

We likely need more data on existing ones.
E.g. description on compute
```
     * @param filter The filter to apply to the operation. Allowed values are 'startswith(instanceView/statuses/code,
     *     'PowerState') eq true', 'properties/latestModelApplied eq true', 'properties/latestModelApplied eq false'.
```
So I think most is using odata $filter $expand. We need first investigate on existing support on odata.